widgets/blueprints/src/TerraformModal.tsx

```diff
@@ -287,14 +289,22 @@ export default function TerraformModal({
         }
     }
 
+    function handleUrlAuthenticationChange(_event: Event, { checked }: CheckboxProps) {
+        setUrlAuthentication(checked);
+        if (!checked) {
+            clearUsername();
+            clearPassword();
+        }
+    }
+
     function handleTemplateUrlBlur() {
-        if (!Stage.Utils.Url.isUrl(templateUrl)) {
+        if (!Stage.Utils.Url.isUrl(templateUrl) || (urlAuthentication && (!username || !password))) {
```

Comment:
```suggestion
    const isAuthenticated = urlAuthentication && (!username || !password);
    if (!Stage.Utils.Url.isUrl(templateUrl) || isAuthenticated ) {
```

We could extract the variable from the `if`, so that we don't have so much logic inside of it
Also, this change would make the function more readable 💪

Reply:
Done, although I used a more accurate variable name.